mvx-esdt-safe/src/register_token.rs

```diff
@@ -13,32 +20,71 @@ pub trait RegisterTokenModule:
     + cross_chain::execute_common::ExecuteCommonModule
     + custom_events::CustomEventsModule
     + multiversx_sc_modules::pause::PauseModule
+    + setup_phase::SetupPhaseModule
 {
     #[payable("EGLD")]
     #[endpoint(registerToken)]
     fn register_token(
         &self,
-        sov_token_id: TokenIdentifier,
-        token_type: EsdtTokenType,
-        token_display_name: ManagedBuffer,
-        token_ticker: ManagedBuffer,
-        num_decimals: usize,
+        hash_of_hashes: ManagedBuffer,
+        token_to_register: SovTokenProperties<Self::Api>,
     ) {
-        self.require_sov_token_id_not_registered(&sov_token_id);
+        let token_hash = token_to_register.generate_hash();
+        if token_hash.is_empty() {
+            self.complete_operation(&hash_of_hashes, &token_hash, Some(ERROR_AT_ENCODING.into()));
+            return;
+        };
 
-        require!(self.has_prefix(&sov_token_id), CANNOT_REGISTER_TOKEN);
-        let issue_cost = self.call_value().egld().clone_value();
+        if self.is_paused() {
+            self.complete_operation(
+                &hash_of_hashes,
+                &token_hash,
+                Some(ESDT_SAFE_STILL_PAUSED.into()),
+            );
 
-        match token_type {
-            EsdtTokenType::Invalid => sc_panic!(INVALID_TYPE),
-            _ => self.handle_token_issue(IssueEsdtArgs {
-                sov_token_id: sov_token_id.clone(),
-                issue_cost,
-                token_display_name,
-                token_ticker,
-                token_type,
-                num_decimals,
-            }),
+            return;
+        }
+
+        if !self.is_setup_phase_complete() {
+            self.complete_operation(
+                &hash_of_hashes,
+                &token_hash,
+                Some(SETUP_PHASE_NOT_COMPLETED.into()),
+            );
+            return;
+        }
+
+        self.lock_operation_hash_wrapper(&hash_of_hashes, &token_hash);
+
```

Comment:
I see an edge case here if the sovereign is malicious. 
We need to make sure here that contract has at least `ISSUE_COST[0.05] EGLD` balance, otherwise just complete_operation with error -> NO deposit back